### PR TITLE
added support for juttle 0.5.0

### PR DESCRIPTION
**Pull request by rlgomes**
few things moved around and then also made sure that eslint knows about
thew new JuttleAdapterAPI global

added check for require -from/-to or -last being set

### CODE REVIEW

**Comment by rlgomes** [comment]
@bkutil @demmer @davidvgalbraith

File: lib/read.js [hunk 1]
```diff
@@ -32,9 +32,17 @@ class ReadGraphite extends AdapterRead {
...
 
         this.name = name;
+        this.from = this.options.from || params.now;
+        this.to = this.options.to || params.now;
     }
 
-    periodicLiveRead() { 
+    _validate_options(options) {
+        if (!options.from && !options.to) {
+            throw new Error('-from, -to, or -last must be specified');
```
Comment (bkutil) [inline lib/read.js]: Maybe Juttle errors and specifying the error code directly would be better - `throw this.compileError('MISSING-TIME-RANGE');`

File: test/graphite.spec.js [hunk 2]
```diff
@@ -35,6 +36,17 @@ describe('graphite-adapter API tests', function () {
...
     });
 
+    it('fails when not provided -from/-to or -last', function() {
+        return check_juttle({
+            program: 'read graphite name="foo"'
+        })
+        .then(function() {
+            throw Error('Previous statement should have failed');
+        }).catch(function(err) {
+            expect(err.message).to.contain('-from, -to, or -last must be specified');
```
Comment (bkutil) [inline test/graphite.spec.js]: ...and then maybe check for the error code here. Should be a bit more future-proof as well, as error codes change less often.

**Comment by davidvgalbraith** [comment]
+1

File: lib/read.js [hunk 3]
```diff
@@ -32,21 +32,29 @@ class ReadGraphite extends AdapterRead {
         }
 
         this.name = name;
+        this.from = this.options.from || params.now;
+        this.to = this.options.to || params.now;
     }
 
-    periodicLiveRead() { 
+    _validate_options(options) {
```
Comment (demmer) [inline lib/read.js]: NIT: Why not camelCase?

Comment (rlgomes) [inline lib/read.js]: copy pasta... I'll fix it before pushing the next release.

**Pull request merged**